src/reducer/database/osm/reconstructor.ts

```diff
@@ -0,0 +1,260 @@
+//#region > Imports
+//#PACKAGE "alasql"
+//## npm install "alasql"@0.5.9
+// Contains a framework for creating a database in the browser
+var alasql = require("alasql");
+//#endregion
+
+//#region > Classes
+/** @class Provides interaction to the database */
+class SOAssembler {
+  static database = new alasql.Database();
+
+  /**
+   * The implementation of the getObjects is necessary for any model!
+   *
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @param Base The class of a OSM model
+   */
+  constructor(private Base: any) {
+    /**
+     * Initialize the table if the necessary statement exists.
+     */
+    if (Base.statements.initialize) {
+      SOAssembler.database.exec(Base.statements.initialize);
+    } else {
+      /**
+       * @todo Error handling
+       */
+    }
+  }
+
+  /**
+   * Create object in database.
+   *
+   * @param fields The data of the object. It has to be in the correct order!
+   * @returns {object} A object of the initialized Base class
+   * @description Creates a entry in the database and returns a object type of
+   *              Base.
+   */
+  create(fields: any) {
+    try {
+      let tablename = new this.Base({}).tableName();
+
+      SOAssembler.database.exec(
+        this.Base.statements.create,
+        Object.values(fields)
+      );
+
+      fields.id = SOAssembler.database.autoval(tablename, "id");
+
+      let obj = new this.Base(fields);
+
+      return obj;
+    } catch (err) {
+      return {
+        success: false,
+        message: err.message,
+      };
+    }
+  }
+
+  /**
+   * Get a object out of the database.
+   *
+   * @param fields The data to get a object by
+   * @returns {object} A object of the initialized Base class
+   * @description Get a object out of the database and convert in to type of
+   *              Base.
+   */
+  get(fields: any) {
+    try {
+      let response = SOAssembler.database.exec(
+        this.Base.statements.get,
+        Object.values(fields)
+      );
+
+      if (response.length === 1) {
+        response = response[0];
+      }
+
+      let obj = new this.Base(response);
+
+      return obj;
+    } catch (err) {
+      return {
+        success: false,
+        message: err.message,
+      };
+    }
+  }
+
+  /**
+   * Get all objects of the table specified by type of Base.
+   *
+   * @deprecated The fields parameter my be useless and should be removed!
+   * @param fields The data to get a object by
+   * @returns {object[]} A list of objects of the initialized Base class
+   */
+  all(...fields: any[]) {
+    try {
+      let response;
+
+      response = SOAssembler.database.exec(this.Base.statements.all, fields);
+      response = response.map((entry: any) => {
+        return new this.Base(entry);
+      });
+
+      return response;
+    } catch (err) {
+      //#LEGACY
+      //#ERROR
+      console.error("OSM reconstructor .all()", err);
```

Comment:
Should this be removed before merging to master, because it is legacy?
```suggestion
```

Reply:
```suggestion
      //#LEGACY
      //#TODO Implement proper error handling
      //#ERROR
      console.error("OSM reconstructor .all()", err);
```